src/collection.js

```diff
@@ -460,15 +462,31 @@ export default class Collection {
    * @return {Promise}
    */
   importChanges(syncResultObject, changeObject) {
-    return Promise.all(changeObject.changes.map(change => {
-      return this._importChange(change);
-    }))
-      .then(imports => {
-        for (let imported of imports) {
-          if (imported.type !== "void") {
-            syncResultObject.add(imported.type, imported.data);
+    const conflicts = [], skipped = [];
+    // Ensure all imports are done within a single transaction
+    const batchResult = this.db.batch(batch => {
+      return Promise.all(changeObject.changes.map(change => {
+        return this._importChange(batch, change).then(importResult => {
+          if (importResult.type === "conflicts") {
+            conflicts.push(importResult.data);
+          } else if (importResult.type === "skipped") {
+            skipped.push(importResult.data);
           }
-        }
+          return importResult;
+        });
+      }));
+    });
+    return batchResult
+      .then(({operations, errors}) => {
```

Comment:
Why not directly `return this.db.batch( ... ).then(({operations, errors}) => {` from line 467?